lib/t/feeds.ex

```diff
@@ -625,4 +629,148 @@ defmodule T.Feeds do
+
+    two_hours_ago_ref = DateTime.add(DateTime.utc_now(), -7200)
+    clear_live_tables(two_hours_ago_ref)
+
+    :ok
+  end
+
+  @doc """
+  Lists ids of today's "newbies live" participants.
+
+  "newbies live" participants are:
+  - new newbies: users who have registered today (they are still newbies even if they've had a call)
+  - TODO old newbies: users who have registered before but haven't had a call
+  - oldies or "midwives": experienced users who explain newbies what all this is about
+  """
+  @spec newbies_list_today_participants :: [Ecto.UUID.t()]
+  def newbies_list_today_participants(reference \\ DateTime.utc_now()) do
+    # maybe Repo.stream later
+    # newbies = new_newbies_q(reference) |> union(^old_newbies_q()) |> Repo.all()
+    newbies = new_newbies_q(reference) |> Repo.all()
+    hardcoded_oldies() ++ newbies
+  end
+
+  defp hardcoded_oldies do
+    Application.get_env(:t, :oldies) || []
+  end
+
+  # TODO move to accounts.ex? maybe later
+  @spec new_newbies_q(DateTime.t()) :: Ecto.Query.t()
+  defp new_newbies_q(reference) do
+    T.Accounts.User
+    |> where([u], is_nil(u.blocked_at))
+    |> where([u], not is_nil(u.onboarded_at))
+    |> where([u], fragment("? + interval '24h' >= ?::timestamp ", u.onboarded_at, ^reference))
+    |> select([u], u.id)
+  end
+
+  # TODO implement properly in the next PR
+  # @spec old_newbies_q :: Ecto.Query.t()
+  # defp old_newbies_q do
+  #   accepted_calls = where(Call, [c], not is_nil(c.accepted_at))
+
+  #   T.Feeds.FeedProfile
+  #   |> where([p], not p.hidden?)
+  #   |> join(:left, [p], c in subquery(accepted_calls),
+  #     on: p.user_id == c.caller_id or p.user_id == c.called_id
+  #   )
+  #   |> where([p, c], is_nil(c.id))
+  #   |> select([p], p.user_id)
+  # end
+
+  @spec is_a_newbies_participant?(Ecto.UUID.t(), DateTime.t()) :: boolean
+  defp is_a_newbies_participant?(user_id, reference) do
+    user_id in hardcoded_oldies() or
+      is_new_newbie?(user_id, reference)
+
+    # or is_old_newbie?(user_id)
+  end
+
+  @spec is_new_newbie?(Ecto.UUID.t(), DateTime.t()) :: boolean
+  defp is_new_newbie?(user_id, reference) do
+    new_newbies_q(reference)
+    |> where(id: ^user_id)
+    |> Repo.exists?()
+  end
+
+  # @spec is_old_newbie?(Ecto.UUID.t()) :: boolean
+  # defp is_old_newbie?(user_id) do
+  #   old_newbies_q()
+  #   |> where(user_id: ^user_id)
+  #   |> Repo.exists?()
+  # end
+
+  @doc """
+  Checks if there is an ongoing "Since Live" event and if that is the case,
+  checks if user_id is a participant.
+
+  Assumes "Since Live" event starts at 19:00 and ends at 21:00 MSK.
+  """
+  @spec newbies_live_now?(Ecto.UUID.t(), DateTime.t()) :: boolean
+  def newbies_live_now?(user_id, reference \\ DateTime.utc_now()) do
+    msk_now = DateTime.shift_zone!(reference, "Europe/Moscow")
+
+    newbie_time? =
+      Time.compare(~T[19:00:00], msk_now) in [:eq, :lt] and
+        Time.compare(~T[21:00:00], msk_now) in [:eq, :gt]
```

Comment:
I'm assuming Since Live for newbies would be as long as the real one, 2 hours, and that it starts at 19:00 MSK.

Reply:
I would say 1 hour is enough